contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,200 @@
+<frontmatter>
+  title: Introduction to Angular
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Introduction to Angular
+
+**Author: [Ronak Lakhotia](https://github.com/RonakLakhotia)**                                                              
+Reviewers: [Tan Heng Yeow](https://github.com/tanhengyeow), [Jacob Li PengCheng](https://github.com/jacoblipech)
+
+## What is Angular?
+
+Angular, according to the [official website](https://angular.io/docs), is described as follows:
+
+>Angular is a platform that makes it easy to build applications with the web. Angular combines declarative templates, dependency injection, end to end tooling, and integrated best practices to solve development challenges. Angular empowers developers to build applications that live on the web, mobile, or the desktop.
+
+Let us understand some key terms starting with what <i>declarative templates</i> mean.
+Consider the example of creating a simple web application as shown below.
+
+![Example](First_example.png)
+
+The application displays the details of a person, that is the id and the name. The view component (i.e., the part users see) is defined in Angular using templates. These templates are nothing but HTML files. Additionally, Angular extends the vocabulary of HTML by allowing you to create your own custom tags.
```

Comment:
Is there any dynamic behavior? If not, the reader will wonder why go to all this trouble to just display a bit of text.

Reply:
> <input text="hero.name" placeholder="name"/>

The above line is unique to how HTML is used in Angular. 

> The `hero.name` used in the HTML code, refers to the name property defined in this component.

Is this what you mean?

Reply:
If the whole thing is just a snippet of static text, why do you even need Angular here?

Reply:
Should I add something more specific to Angular? 
Initially I thought of adding the following code snippet

```
<h2>{{hero.name}} Details</h2>
<div><span>id: </span>{{hero.id}}</div>
<div>
  <label>name:
    <input [(ngModel)]="hero.name" placeholder="name"/>
  </label>
</div>
```
 and explain the use of {{}} -> interpolation and terms like ngModel. But would that be too much for the begining of the article?

Reply:
> If the whole thing is just a snippet of static text, why do you even need Angular here?

In the example I used in the article, the Name of the person is dynamically updated. It is not standard HTML and is relevant to Angular. So if the name of the person is changed in the component file, the name is automatically updated in the view. This is due to the property binding in Angular

Reply:
Sure, when I update the typescript file, the page changes. I can just as well update the HTML file and get the same result. We don't need Angular for that. In this example, Angular makes a simple thing unnecessarily complex. On the other hand, If the user can enter a name in the browser and something else in the page changes based on that value, that's something you can't do with plain HTML.
i.e., you need an example where Angular actually improves the situation and at the same time need to keep it as simple as possible.